akka-http-core/src/main/java/akka/http/javadsl/model/headers/SameSite.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.javadsl.model.headers;
+
+/**
+ * The Cookie SameSite attribute as defined by <a href="https://tools.ietf.org/html/draft-ietf-httpbis-cookie-same-site-00">RFC6265bis</a>
+ * and <a href="https://tools.ietf.org/html/draft-west-cookie-incrementalism-00">Incrementally Better Cookies</a>.
+ */
+public enum SameSite {
+    Strict,
+    Lax,
+    // SameSite.None is different from not adding the SameSite attribute in a cookie.
+    // - Cookies without a SameSite attribute will be treated as SameSite=Lax.
+    // - Cookies for cross-site usage must specify `SameSite=None; Secure` to enable inclusion in third party
+    //   context. We are not enforcing `; Secure` when `SameSite=None`, but users should.
```

Comment:
What we have done before is to add a `require()` to a model class if there are certain constraints. Not sure if we would do it here, though, at least not as long there's no proper spec.

Reply:
There is a draft spec though: https://tools.ietf.org/html/draft-west-cookie-incrementalism-00

It says _cookies that explicitly assert "SameSite=None" in order to enable cross-site delivery should also be marked as "Secure"_.

With the current state of this PR, it is up to users to correctly call `withSameSite(SameSite.None)` *and* `withSecure(true)` for that cookie. Using a `require` seems like the correct approach to me.

WDYT?

Reply:
I also wonder if we should set `secure=true` automatically when calling `withSameSite(SameSite.None)`. Maybe a combination of that and the `require` to handle the case where users (accidentally, I hope) call `withSameSite(SameSite.None).withSecure(false)`.

Reply:
Ah, missed that. I agree both suggestions makes sense. We should have a test that makes sure that this `require` fails the response in a reasonable way.

Reply:
I realized that it won't be possible to do this check. Both cookies below are valid:

```
Set-Cookie: name=123; Secure; SameSite=None
```

```
Set-Cookie: name=123; SameSite=None; Secure
```

But the second one sets `SameSite` first, and the parser will then try to create a `HttpCookie` with that, but fails because `secure=false` since it was not parsed yet. It feels to me that using a `require` is good when you are checking an attribute that does not relate to another one.

WDYT?

Reply:
Sounds fine to me to skip it then.

Reply:
Ah, yes, if it's built incrementally, then `require` won't work, so the only thing we could do is set `Secure` automatically when we see `SameSite=None` and only fail when trying to remove `Secure` when `SameSite=None` is set as well. But not sure if we should go for that right now?